dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/objectbundle/hooks/ProgramObjectBundleHook.java

```diff
@@ -76,4 +114,29 @@ private void syncSharingForEventProgram( Program program )
         programStage.setUser( program.getUser() );
         sessionFactory.getCurrentSession().update( programStage );
     }
+    
+    private void addProgramInstance( Program program )
+    {
+        if ( getProgramInstancesCount( program ) == 0 )
+        {
+            ProgramInstance pi = new ProgramInstance();
+            pi.setEnrollmentDate( new Date() );
+            pi.setIncidentDate( new Date() );
+            pi.setProgram( program );
+            pi.setStatus( ProgramStatus.ACTIVE );
+            pi.setStoredBy( "system-process" );
+
+            this.programInstanceService.addProgramInstance( pi );
+        }
+    }
+    
+    private int getProgramInstancesCount( Program program )
+    {
+        return programInstanceService.getProgramInstances( program, ProgramStatus.ACTIVE ).size();
```

Comment:
Can you try and see if `countProgramInstances` works better? this will use hql, and doesn't require loading up the full hibernate graph